peer/network.go

```diff
@@ -190,7 +190,8 @@ func (n *network) sendAppRequest(nodeID ids.NodeID, request []byte, responseHand
 
 	// Send app request to [nodeID].
 	// On failure, release the slot from [activeAppRequests] and delete request from [outstandingRequestHandlers]
-	if err := n.appSender.SendAppRequest(context.TODO(), nodeIDs, requestID, request); err != nil {
+	// TODO: treat error from  n.appSender.SendAppRequest as fatal
```

Comment:
Can we remove these comments?

This is marked as "should be treated as fatal" in AvalancheGo, but there are other errors that could be returned here that should not be treated as fatal.

So... I think we will likely continue treating this as non-fatal here, since treating it as fatal would probably involve a special cased error or extra return value, which doesn't seem worth it.

If there's an alternative way you'd suggest treating this as fatal, then it would make sense to add the TODO, otherwise I think we should remove this.

Reply:
In avalanchego's `sync` package we have an error value for this case and check for it (see [here](https://github.com/ava-labs/avalanchego/blob/master/x/sync/client.go#L300) and [here](https://github.com/ava-labs/avalanchego/blob/master/x/sync/network_client.go#L266)).

I think that we should try to honor the invariant on the `AppSender`, but I don't think it's _strictly_ required, so if you don't want to do something like in avalanchego then I can remove these comments.

Reply:
Can we remove this? I think that makes the code harder to follow